src/plugins/es_ui_shared/static/forms/components/fields/card_radio_group_field.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { Fragment, ReactChildren } from 'react';
+import { EuiFormRow, EuiSpacer, EuiCheckableCard, useGeneratedHtmlId } from '@elastic/eui';
+
+import { FieldHook, getFieldValidityAndErrorMessage } from '../../hook_form_lib';
+
+interface Props {
+  field: FieldHook;
+  options: Array<{ label: string; value: string; children: ReactChildren }>;
+  euiFieldProps?: Record<string, any>;
+  idAria?: string;
+  [key: string]: any;
+}
+
+export const CardRadioGroupField = ({
+  field,
+  options,
+  euiFieldProps = {},
+  idAria,
+  ...rest
+}: Props) => {
+  const { isInvalid, errorMessage } = getFieldValidityAndErrorMessage(field);
+  const radioGroupId = useGeneratedHtmlId({ prefix: 'radioGroup' });
+
+  return (
+    <EuiFormRow
+      label={field.label}
+      labelType="legend"
+      helpText={typeof field.helpText === 'function' ? field.helpText() : field.helpText}
+      error={errorMessage}
+      isInvalid={isInvalid}
+      fullWidth
+      describedByIds={idAria ? [idAria] : undefined}
+      {...rest}
+    >
+      <>
+        {options.map(({ label, value, children }) => (
+          <Fragment key={`${radioGroupId}-${value}`}>
+            <EuiCheckableCard
```

Comment:
How do you feel about providing the user a way to pass in a data-test-subj to the checkable cards? Otherwise it seems they wont be able to easily assert against them